cohort-engine/src/main/java/com/ibm/cohort/engine/InsecureApacheRestfulClientFactory.java

```diff
@@ -0,0 +1,186 @@
+import ca.uhn.fhir.rest.client.api.Header;
+import ca.uhn.fhir.rest.client.api.IHttpClient;
+import ca.uhn.fhir.rest.client.impl.RestfulClientFactory;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.config.Registry;
+import org.apache.http.config.RegistryBuilder;
+import org.apache.http.conn.socket.ConnectionSocketFactory;
+import org.apache.http.conn.socket.PlainConnectionSocketFactory;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.client.ProxyAuthenticationStrategy;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A copy of ApacheRestfulClientFactory that ignores SSL CA and hostname verification.
+ *
+ * THIS SHOULD NOT BE USED IN PRODUCTION
+ */
+public class InsecureApacheRestfulClientFactory extends RestfulClientFactory {
+
+	private HttpClient myHttpClient;
+	private HttpHost myProxy;
+
+	/**
+	 * Constructor
+	 *
+	 * @param theContext
+	 *            The context
+	 */
+	public InsecureApacheRestfulClientFactory(FhirContext theContext) {
+		super(theContext);
+	}
+
+	@Override
+	protected synchronized ApacheHttpClient getHttpClient(String theServerBase) {
+		return new ApacheHttpClient(getNativeHttpClient(), new StringBuilder(theServerBase), null, null, null, null);
+	}
+
+	@Override
+	public synchronized IHttpClient getHttpClient(StringBuilder theUrl, Map<String, List<String>> theIfNoneExistParams,
+												  String theIfNoneExistString, RequestTypeEnum theRequestType, List<Header> theHeaders) {
+		return new ApacheHttpClient(getNativeHttpClient(), theUrl, theIfNoneExistParams, theIfNoneExistString, theRequestType,
+				theHeaders);
+	}
+
+	public HttpClient getNativeHttpClient() {
+		if (myHttpClient == null) {
+
+			PoolingHttpClientConnectionManager connectionManager = createInsecureHttpConnectionManager(5000, TimeUnit.MILLISECONDS);
+			connectionManager.setMaxTotal(getPoolMaxTotal());
+			connectionManager.setDefaultMaxPerRoute(getPoolMaxPerRoute());
+
+			//TODO: Use of a deprecated method should be resolved.
```

Comment:
stray `todo`

Reply:
That todo is actually copied from [HAPI FHIR](https://github.com/hapifhir/hapi-fhir/blob/cbb16ce3affd3fc53dcbfe98dd3181644fe68604/hapi-fhir-client/src/main/java/ca/uhn/fhir/rest/client/apache/ApacheRestfulClientFactory.java#L93).